pageserver/src/tenant/timeline.rs

```diff
@@ -476,6 +476,15 @@ impl GcInfo {
     pub(crate) fn min_cutoff(&self) -> Lsn {
         self.cutoffs.select_min()
     }
+
+    pub(super) fn insert_child(&mut self, child_id: TimelineId, child_lsn: Lsn) {
+        self.retain_lsns.push((child_lsn, child_id));
+        self.retain_lsns.sort_by_key(|i| i.0);
```

Comment:
sounds like BTreeMap is the right data structure for retain_lsns, may refactor it in the future.